tests/mocha/utils_test.js

```diff
@@ -15,6 +15,201 @@ suite('Utils', function() {
     }
   });
 
+  test('tokenizeInterpolation', function() {
+    var tokens = Blockly.utils.tokenizeInterpolation('');
+    assertArrayEquals('Null interpolation', [], tokens);
+
+    tokens = Blockly.utils.tokenizeInterpolation('Hello');
+    assertArrayEquals('No interpolation', ['Hello'], tokens);
+
+    tokens = Blockly.utils.tokenizeInterpolation('Hello%World');
+    assertArrayEquals('Unescaped %.', ['Hello%World'], tokens);
+
+    tokens = Blockly.utils.tokenizeInterpolation('Hello%%World');
+    assertArrayEquals('Escaped %.', ['Hello%World'], tokens);
+
+    tokens = Blockly.utils.tokenizeInterpolation('Hello %1 World');
+    assertArrayEquals('Interpolation.', ['Hello ', 1, ' World'], tokens);
+
+    tokens = Blockly.utils.tokenizeInterpolation('%123Hello%456World%789');
+    assertArrayEquals('Interpolations.', [123, 'Hello', 456, 'World', 789], tokens);
+
+    tokens = Blockly.utils.tokenizeInterpolation('%%%x%%0%00%01%');
+    assertArrayEquals('Torture interpolations.', ['%%x%0', 0, 1, '%'], tokens);
+
+    Blockly.Msg = Blockly.Msg || {};
```

Comment:
Split this into a few chunks (splitting it per individual call to tokenize feels like overkill):
- Basic interpolation (to here)
- Message references (to line 55)
- Message reference errors (56-83)

Reply:
Done.